python-pytest.el

```diff
@@ -116,44 +116,46 @@ When non-nil only ‘test_foo()’ will match, and nothing else."
 
-;;;###autoload (autoload 'python-pytest-popup "python-pytest" nil t)
-(magit-define-popup python-pytest-popup
+;;;###autoload (autoload 'python-pytest-dispatch "python-pytest" nil t)
+(define-transient-command python-pytest-dispatch ()
   "Show popup for running pytest."
-  'python-pytest
-  :switches
-  '((?c "color" "--color" t)
-    (?d "run doctests" "--doctest-modules")
-    (?f "failed first" "--failed-first")
-    (?l "show locals" "--showlocals")
-    (?p "debug on error" "--pdb")
-    (?q "quiet" "--quiet")
-    (?s "do not capture output" "--capture=no")
-    (?t "do not cut tracebacks" "--full-trace")
-    (?v "verbose" "--verbose")
-    (?w "very verbose" "-vv")
-    (?x "exit after first failure" "--exitfirst"))
-  :options
-  '((?k "only names matching expression" "-k")
-    (?m "only marks matching expression" "-m")
-    (?t "traceback style" "--tb=" python-pytest--choose-traceback-style)
-    (?x "exit after N failures or errors" "--maxfail="))
-  :actions
-  '("Run tests"
-    (?t "Test all" python-pytest)
-    (?r "Repeat last test run" python-pytest-repeat)
-    (?x "Test last-failed" python-pytest-last-failed)
-    "Run tests for specific files"
-    (?f "Test file (dwim)" python-pytest-file-dwim)
-    (?F "Test this file" python-pytest-file)
-    (?m "Test multiple files" python-pytest-files)
-    "Run tests for current function/class"
-    (?d "Test def/class (dwim)" python-pytest-function-dwim)
-    (?D "Test this def/class" python-pytest-function))
-  :max-action-columns 3
-  :default-action 'python-pytest-repeat)
+  :man-page "pytest"
+  :incompatible '(("--verbose" "--verbose --verbose"))
+  :value '("--color")
+  ["Switches"
+   ("-c" "color" "--color")
+   ("-d" "run doctests" "--doctest-modules")
+   ("-f" "failed first" "--failed-first")
+   ("-l" "show locals" "--showlocals")
+   ("-p" "debug on error" "--pdb")
+   ("-q" "quiet" "--quiet")
+   ("-s" "do not capture output" "--capture=no")
+   ("-t" "do not cut tracebacks" "--full-trace")
+   ("-v" "verbose" ("-v" "--verbose"))
+   ("-w" "very verbose" ("-vv" "--verbose --verbose"))
+   ("-x" "exit after first failure" "--exitfirst")]
+  ["Options"
+   ("=k" "only names matching expression" "-k=")
+   ("=m" "only marks matching expression" "-m=")
+   (python-pytest-dispatch:--tb)
+   ("=x" "exit after N failures or errors" "--maxfail=")]
+  [["Run tests"
+    ("t" "Test all" python-pytest)
+    ("r" "Repeat last test run" python-pytest-repeat)
+    ("x" "Test last-failed" python-pytest-last-failed)]
+   ["Run tests for specific files"
+    ("f" "Test file (dwim)" python-pytest-file-dwim)
+    ("F" "Test this file" python-pytest-file)
+    ("m" "Test multiple files" python-pytest-files)]
+   ["Run tests for current function/class"
+    ("d" "Test def/class (dwim)" python-pytest-function-dwim)
+    ("D" "Test this def/class" python-pytest-function)]])
+
+(define-obsolete-function-alias 'python-pytest-popup
+  'python-pytest-dispatch "python-pytest 0.3.2")
```

Comment:
```suggestion
(define-obsolete-function-alias 'python-pytest-popup 'python-pytest-dispatch)
```

i don't think the [`when`](https://www.gnu.org/software/emacs/manual/html_node/elisp/Obsolete-Functions.html#Obsolete-Functions) argument contains the right value, but since it's an optional argument it can just be left out i guess?

Reply:
I went for version one up over what is it `melpa-unstabel`. But I think it can be any text as it has no semantic meaning. How about `python-pytest as of Aug/2020` for `when`? If you don't like it, I'm happy to remove as well.

Reply:
the package and readme actually list 1.0.0 as the version but apparently i never bother to push a `git tag` for it... oops :see_no_evil: 

let's leave it out altogether, the version numbers are pretty much meaningless anyway. practically, anything not being the tip of the main branch (`git master` for this project) is abandonware anyway.